python/cuda_cccl/tests/compute/examples/iterator/discard_iterator_basic.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2024-2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# example-begin
+"""
+Example showing how to use DiscardIterator.
+"""
+
+import cupy as cp
+import numpy as np
+
+import cuda.compute
+from cuda.compute import (
+    DiscardIterator,
+    OpKind,
+)
+
+# Prepare the input and output arrays.
+h_in_keys = np.array([1, 1, 2, 3, 3, 7, 8, 8], dtype="int32")
+d_in_keys = cp.asarray(h_in_keys)
```

Comment:
Nit: since `h_in_keys` is not used in the verification later, let's just skip it:
```suggestion
d_in_keys = cp.asarray([1, 1, 2, 3, 3, 7, 8, 8], dtype=cp.int32)
```